sql/core/src/test/scala/org/apache/spark/sql/execution/command/AlterTableDropPartitionSuiteBase.scala

```diff
@@ -144,4 +145,14 @@ trait AlterTableDropPartitionSuiteBase extends QueryTest with DDLCommandTestUtil
       checkPartitions(t)
     }
   }
+
+  test("SPARK-33591: null as a partition value") {
+    withNamespaceAndTable("ns", "tbl") { t =>
+      sql(s"CREATE TABLE $t (col1 INT, p1 STRING) $defaultUsing PARTITIONED BY (p1)")
+      sql(s"ALTER TABLE $t ADD PARTITION (p1 = null)")
```

Comment:
Does it work for hive catalog?

Reply:
This is common test suite for all catalogs. Yes, it works for Hive external catalog too.

Reply:
hmm, don't we have a check to fail null partition value for ADD PARTITION with hive catalog? https://github.com/apache/spark/pull/30538/files#diff-9dd0899e5406230aeff96654432da54f35255f6dc60eecb87264a5c508a8c826R1193

Reply:
> don't we have a check to fail null partition value for ADD PARTITION with hive catalog

For hive catalog, we convert `null` to `DEFAULT_PARTITION_NAME` by `ExternalCatalogUtils.convertNullPartitionValues` from `toMetaStorePartitionSpec`. The last one is invoked in all functions in `HiveExternalCatalog` that makes deal with partition specs.